src/commands/importFromRegistry.ts

```diff
@@ -0,0 +1,168 @@
+            username: (username === undefined || username === "" ? undefined : username),
+            webServer: {
+              host: getProperty("WebServerAddress") || getProperty("Address"),
+              pathPrefix: getProperty("WebServerInstanceName"),
+              port: parseInt(getProperty("WebServerPort"), 10),
+              scheme: usesHttps ? "https" : "http",
+            },
+          };
+        }
+      });
+
+      if (serversMissingUsernames.length) {
+        let serverName = serversMissingUsernames[0];
+        let username = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: "username",
+          prompt: `Username for server '${serverName}'`,
+          validateInput: ((value) => {
+              return value.length > 0 ? "" : "Mandatory field";
+          }),
+        }, cancellationToken);
+        if (cancellationToken.isCancellationRequested || username === undefined) {
+          return;
+        }
+        if (serversMissingUsernames.length > 1) {
+          const items = [
+            `Use username '${username}' for all servers`,
+            `Enter usernames individually for ${serversMissingUsernames.length - 1} more server(s)`,
+            `Cancel`].map((label) => {
+              return { label };
+            });
+          const result = await vscode.window.showQuickPick(items, {
+            canPickMany: false,
+            ignoreFocusOut: true,
+          }, cancellationToken);
+          if (cancellationToken.isCancellationRequested || result === undefined || result.label === items[2].label) {
+            return;
+          } else if (result.label === items[0].label) {
+            for (serverName of serversMissingUsernames) {
+              serverDefinitions[serverName].username = username;
+            }
+          } else {
+            for (serverName of serversMissingUsernames) {
+              username = await vscode.window.showInputBox({
+                ignoreFocusOut: true,
+                prompt: `Username for server '${serverName}'`,
+                validateInput: ((value) => {
+                    return value.length > 0 ? "" : "Mandatory field";
+                }),
+                value: username,
+              }, cancellationToken);
+              if (cancellationToken.isCancellationRequested || username === undefined) {
+                return;
+              }
+              serverDefinitions[serverName].username = username;
+            }
+          }
+        }
+      }
+
+      let reusePassword;
+      let password;
+      for (const serverName of newServerNames) {
+        if (!reusePassword) {
+          password = await vscode.window.showInputBox({
+            ignoreFocusOut: true,
+            password: true,
+            placeHolder: "Password to store in keychain",
+            prompt: `For connection to InterSystems server '${serverName}'
+              as ${serverDefinitions[serverName].username}`,
```

Comment:
![image](https://user-images.githubusercontent.com/6726799/101466438-0f724c00-3939-11eb-8f4b-3ec5b74d632c.png)

"undefiined" should have been the username I entered at the previous prompt. In the 1-server-only case this doesn't get set into `serverDefinitions[serverName].username` because of an earlier bug (see a previous comment).